ui/src/components/Modal/__tests__/Modal.test.ts

```diff
@@ -0,0 +1,50 @@
+import { mount, shallowMount } from "@vue/test-utils";
+import Modal from "../Modal.vue";
+import { CreateElement, VNode } from "vue";
+
+describe("Modal", () => {
+  const title = "Modal Title";
+  const wrapper = mount(Modal, {
+    propsData: {
+      title,
+      active: false
+    },
+    slots: {
+      actions: {
+        render(h: CreateElement): VNode {
+          return h("div", {}, "Action 1");
+        }
+      }
+    }
+  });
+
+  it("renders correctly", () => {
+    expect(wrapper.element).toMatchSnapshot();
+  });
+
+  it("renders modal title correctly", () => {
+    expect(wrapper.find(".fd-modal__title").text()).toBe(title);
+  });
+
+  it("has modal action", () => {
+    expect(wrapper.find(".fd-modal__actions").exists()).toBe(true);
+  });
+
+  it("is active when set as active", () => {
+    wrapper.setProps({ active: true });
+    expect(wrapper.vm.$data.isActive).toBe(true);
+    expect(wrapper.isVisible()).toBe(true);
+  });
+
+  it("closed correctly", async () => {
+    const button = wrapper.find(".fd-modal__close");
+    button.trigger("click");
+    wrapper.setProps({ active: false });
```

Comment:
why do we need to set this prop to false, will it set to false after clicking the button

Reply:
yes, that is because the click event only emit "close" and "update:active" and it does not set the active to false.

Reply:
i think we should add "this.isActive = false" in the private close()

Reply:
I think adding `this.isActive = false` will fix the issues @suwarnoong has down below (the sync-issue).

If your test would look exactly like the example then the sync issues would not be there.

This is because the example is going though the vue template compiler which is causing to catch the `update:active`-event and updates the state.

Our modal breaks down once the consumer is doing this:


```html
<FdModal :active="false">…</FdModal>
```
and not

```html
<FdModal :active.sync="active">…</FdModal>
```

Meaning: Our modal relies upon the fact that the consuming app is either using .sync or listening for the update-event accordingly + is binding the active-prop.

For this PR I suggest to do what @xujing-shen suggests but make a mental note (=> or an issue) to have a closer look at the modal component again. I think FdModal should me more resilient and support different modes of operation...

Reply:
Note to myself: This is maybe a great question for our Q/A with Chris.

Reply:
Adding `this.isActive = false` in the `close()` will only close the FdModal for the very first time. But `isActive` value (in Modal.vue) is not propagated to parent (Example.vue). Meaning, the parent variable will still be `true`. Subsequent, the `showModal` will not trigger the `watch` for the `active`.

![image](https://user-images.githubusercontent.com/48011859/54006861-1a171480-419a-11e9-9823-5fde1d9dc092.png)
 
If the consumer doesn't use `.sync`, alternatively, they could listen on `@close` event and set the parent variable to false.

But, in my personal opinion, perhaps, we should only let parent to hide/show through `hide()' `show()` methods instead of just a variable. But this implementation will be different with fundamental-react. Are we allow to have different implementation?

```
<Modal onClose={this.showHideModal} show={this.state.bShowInfoModal}>
  ...
</Modal>

// show / hide information modal
showHideModal = () => {
  this.setState(prevState => ({
    bShowInfoModal: !prevState.bShowInfoModal
  }));
};
```

Reply:
You are correct. If the consume is not using `:visible.sync` then it won't work. This is what I meant by saying:

> Meaning: Our modal relies upon the fact that the consuming app is either using .sync or listening for the update-event accordingly + is binding the active-prop.

In the end, `:active.sync="active"` is syntactic sugar for:

```html
<FdModal @update:active="active = $event" :active="active" … >
```

I think, when I wrote the code for `FdModel` I was heavily influenced by: https://element.eleme.io/#/en-US/component/dialog.

We are allowed to have different implementations. After all: We are all using different frameworks which have different features and philosophies. :)

With regards to your "methods-idea": Personally I think calling a method is less convenient than to have a `sync-prop`. It is also not less prone to errors. But: I think there is no right and wrong and there is no best way to do things. Every solution has pro and cons. I also think that bootstrap-vue has a different but very well designed solution:

https://bootstrap-vue.js.org/docs/components/modal#toggle-modal-visibility

Their solution is more or less what I had in mind for `FdPopper`. :D The world is so small.

If you want to add an `open(…)` method: Why not? :)